diff -ur original/usr/share/lipstick-jolla-home-qt5/compositor.qml patched/usr/share/lipstick-jolla-home-qt5/compositor.qml
--- original/usr/share/lipstick-jolla-home-qt5/compositor.qml	2017-01-26 05:47:10.000000000 +0530
+++ patched/usr/share/lipstick-jolla-home-qt5/compositor.qml	2017-01-29 05:49:55.000000000 +0530
@@ -71,6 +71,7 @@
     property alias launcherLayer: launcherLayerItem
     property alias homeLayer: homeLayerItem
     property alias appLayer: appLayerItem
+    property alias appletLayer: appletLayer
     property alias lockScreenLayer: lockScreenLayerItem
     property alias dialogLayer: dialogLayerItem
     property alias alarmLayer: alarmLayerItem
@@ -131,7 +132,7 @@
     property int homeOrientation: Qt.PortraitOrientation
 
     property bool directRendering: appLayer.active && !homeLayer.visible && topmostWindow && topmostWindow.window && !topmostWindow.window.isInProcess
-                                   && !(dialogLayer.contentItem.children.length > 1 || alarmLayer.contentItem.children.length > 1 || notificationLayer.contentItem.children.length || launcherLayer.visible || overlayLayer.contentItem.childrenRect.width || overlayLayer.contentItem.childrenRect.height)
+                                   && !(dialogLayer.contentItem.children.length > 1 || alarmLayer.contentItem.children.length > 1 || appletLayer.contentItem.children.length > 1 || notificationLayer.contentItem.children.length || launcherLayer.visible || overlayLayer.contentItem.childrenRect.width || overlayLayer.contentItem.childrenRect.height)
     fullscreenSurface: directRendering ? topmostWindow.window.surface : null
     screenOrientation: orientationLock == "portrait" ? Qt.PortraitOrientation : (orientationLock == "landscape" ? Qt.LandscapeOrientation : QtQuick.Screen.primaryOrientation)
     keymap: Keymap {
@@ -348,6 +349,11 @@
             break
         }
 
+        if(window.category === "applet") {
+            appletLayer.raised = true
+            window.takeFocus()
+        }
+
         focusTransientInLocalScope(window, transientWindow)
     }
 
@@ -423,6 +429,10 @@
         else if (dialogLayer.window) windows.push(dialogLayer.window.window)
         else if (topmostWindow) windows.push(topmostWindow.window)
 
+        var applets = appletLayer.contentItem.children
+        for (var ii = 0; ii < applets.length; ++ii)
+            windows.push(applets[ii].window)
+
         var overlays = overlayLayer.contentItem.children
         for (var ii = 0; ii < overlays.length; ++ii)
             windows.push(overlays[ii].window)
@@ -976,6 +986,7 @@
                 }
             }
 
+
             Item {
                 id: powerKeyStackBackground
                 anchors.fill: parent
@@ -1031,6 +1042,19 @@
             }
         }
 
+        Item {
+            id: appletLayer
+
+            property alias contentItem: appletLayer
+            property alias overlayItem: appletLayer
+            property bool raised
+
+            anchors.fill: root
+
+            onChildrenChanged: updateWindows()
+            visible: !root.deviceIsLocked && !root.screenIsLocked
+        }
+
         OverlayLayer {
             id: overlayLayer
         }
@@ -1096,6 +1120,7 @@
         var isShutdownWindow = window.isInProcess && window.title == "Shutdown"
         var isPowerWindow = window.isInProcess && window.category == "power"
         var isEventsWindow = window.isInProcess && window.category == "events"
+        var isAppletWindow = window.category == "applet"
         var isDialogWindow = window.category == "dialog"
         var isNotificationWindow = window.category == "notification"
         var isOverlayWindow = window.category == "overlay"
@@ -1130,6 +1155,9 @@
         } else if (isDialogWindow) {
             parent = dialogLayer
             type = WindowType.Dialog
+        } else if (isAppletWindow) {
+            parent = appletLayer
+            parentItem = appletLayer.contentItem
         } else if (isNotificationWindow) {
             parent = notificationLayer
             if (!onlyCurrentNotificationAllowed) {
@@ -1153,7 +1181,7 @@
             return
         }
 
-        window.focusOnTouch = !window.isInProcess && !isOverlayWindow && !isNotificationWindow
+        window.focusOnTouch = !window.isInProcess && !isOverlayWindow && !isNotificationWindow && !isAppletWindow
 
         // reparenting already handled (window is transient to another window)
         if (isApplicationWindow && window.parent !== root.contentItem) {
@@ -1195,6 +1223,7 @@
             // Force notification of clipboard changes to keyboard
             window.surface.updateSelection()
             root.clipboard.dataChanged.connect(window.surface.updateSelection)
+        } else if (isAppletWindow) {
         } else if (isNotificationWindow) {
         } else if (isAlarmWindow) {
             if (!root.visible) {
@@ -1267,6 +1296,11 @@
             homeLayerItem.partnerWindowLowered(window.userData)
             break
         }
+
+        if(window.category == "applet") {
+            clearKeyboardFocus()
+            appletLayer.raised = false
+        }
     }
     onWindowHidden: {
         if (window.category == "cover" || window.category == "silica-cover")
diff -ur original/usr/share/lipstick-jolla-home-qt5/layers/OverlayLayer.qml patched/usr/share/lipstick-jolla-home-qt5/layers/OverlayLayer.qml
--- original/usr/share/lipstick-jolla-home-qt5/layers/OverlayLayer.qml	2017-01-22 00:41:28.000000000 +0530
+++ patched/usr/share/lipstick-jolla-home-qt5/layers/OverlayLayer.qml	2017-01-29 05:48:51.000000000 +0530
@@ -13,6 +13,13 @@
     Connections {
         target: root
         onActiveFocusItemChanged: {
+            // See if the applet layer needs focus
+            if(root.appletLayer.raised) {
+                overlayLayer.parent = root.appletLayer
+                overlayLayer.visible = true
+                return
+            }
+
             // Search for the layer of the focus item
             var focusedLayer = root.activeFocusItem
             while (focusedLayer && focusedLayer.__compositor_is_layer === undefined)
